### PR TITLE
Use OptionalExclude box for excluded reviewers - update UI to conform to newest wireframes

### LINKED ISSUE #509: Update input for excluding reviewers
**Product recs**

- Clicking on 'Would you like to exclude a reviewer' should generate a hide/show panel that contains inputs for all three potential exclusions at once

- There should be one reason to account for all exclusions, which fits the data model

- Clicking the cross should hide the panel

- If the panel has been opened, there should be input in both reviewer 1 name and reviewer 1 email text boxes (validation message: 'Name required' and 'Email required')

- There should be no validation against the optional fields UNLESS one has input in, i.e. if user has entered the name of optional reviewer 2 they must also include an email and vice versa

- ~To ensure consistency throughout the application, please update labels of optional fields to "XXX (optional)" (e.g. "Excluded reviewer 1 name (optional)")~ - separate issue has been raised for this #508 

**Designs**

wizard-mobile-reviewers 2x

### PR DESCRIPTION
**Pull request by jsms90**
#### What does this PR do?
- change graphql, yup & joi schema so that there is a single reason against all excluded/opposed reviewers
- change UI to conform to new wireframes - use `OptionalExclude` panel for reviewers (as we're doing for editors)
- remove superfluous `MoreButton` component
- refactor to remove `FormSections` file - move Declaration checkbox & Reviewers input fields into `EditorsContent` file with the rest

#### Any relevant tickets
Fixes #509 

#### How has this been tested?
- added `opposedReviewers` & `opposedReviewerReason` to schema test file
  - names & reason to "valid data"
  - names but no reason to "invalid data"

### CODE REVIEW

**Comment by jsms90** [comment]
[`ValueCheckbox`](https://github.com/elifesciences/elife-xpub/pull/531/files#diff-d6b938ac5ec1eab7c71a4179e14b0fd2R40) now exists in 2 places in our codebase. So I'm just doing a teeny refactor to pop that into the `atoms` & will push it to this PR. But @damnhipster I know you're in the middle of reviewing this, so I won't do it just yet?

File: app/components/pages/SubmissionWizard/steps/Editors/EditorsContent.js [hunk 1]
```diff
@@ -300,39 +301,75 @@ class EditorsContent extends React.Component {
...
 
-        <Declaration />
+        <ValidatedField
+          component={ValueCheckbox}
+          label="I declare that, to the best of my knowledge, these experts have no conflict of interest"
+          name="suggestionsConflict"
+        />
       </React.Fragment>
     )
   }
```
Comment (damnhipster) [inline app/components/pages/SubmissionWizard/steps/Editors/EditorsContent.js]: This file seems to be getting quite large, and the render functions is about 200 lines long. I suggest we pull out some of the variables on the components into arrays of objects and map them onto the components.

e.g.
```
const sections = [
  {
    heading: 'Suggest Senior Editors',
    type: 'include',
    test-id: 'suggested-senior-editors',
    peoplePicker: {
      modal: 'suggestedSeniorEditors',
      selection: 'suggestedSeniorEditors',
      options: seniorEditors,
      title: 'senior editors'
    }
  },
  {
    heading: 'Exclude a Senior Editor',
    type: 'exclude',
    test-id: 'opposed-senior-editors',
    peoplePicker: {
      modal: 'opposedSeniorEditors',
      selection: 'opposedSeniorEditors',
      options: seniorEditors,
      title: 'senior editors'
    }
  },
  ...
]

sections.map((section) => {
  switch(section.type) {
    case 'include':
      <Box>
        <FormH3>{section.heading}</FormH3>
        <Box data-test-id={section.test-id}>
          <PeoplePicker {...section.peoplePicker}
        </Box>
      </Box>
      break;
    case 'exclude':
      <OptionalExclude>
        <FormH3>{section.heading}</FormH3>
        <Box data-test-id={section.test-id}>
          <PeoplePicker {...section.peoplePicker}
        </Box>
      </OptionalExclude>
      break;
  }
});

```

Comment (damnhipster) [reply app/components/pages/SubmissionWizard/steps/Editors/EditorsContent.js]: Happy to move this onto a seperate issue

Comment (jsms90) [reply app/components/pages/SubmissionWizard/steps/Editors/EditorsContent.js]: Completely agreed. Dying to refactor :heart_eyes: but yeah, figured large-scale changes were outside the scope of this issue. Just wanted to get rid of `FormSection.js` as it wasn't really making sense any more.

As for the way in which we refactor, let's have a bit of a chat before raising an issue?

**Comment by jsms90** [comment]
@damnhipster Ok just pushed up the addition of `ControlledCheckbox` to the atoms folder

**Pull request merged**